doc/zeek.rst

```diff
@@ -428,6 +436,32 @@ processing:
 
 .. include:: /autogen/zeek-functions.spicy
 
+.. _zeek_dpd:
+
+Dynamic Protocol Detection (DPD)
+================================
+
+Spicy protocol analyzers support Zeek's *Dynamic Protocol Detection*
+(DPD), i.e., analysis independent of any well-known ports. To use that
+with your analyzer, add two pieces:
+
+1. A `Zeek signature
+   <https://docs.zeek.org/en/current/frameworks/signatures.html>`_ to
+   activate your analyzer based on payload patterns. Just like with
+   any of Zeek's standard analyzers, a signature can activate a Spicy
+   analyzer through the ``enable "<name>"`` keyword. The name of the
+   analyzer comes out of the EVT file: it is the ``ANALYZER_NAME``
+   with the double colons replaced with an underscore (e.g.,
+   ``spicy::HTTP`` turns into ``enable "spicy_HTTP"``.
+
+2. You should call ``zeek::confirm_protocol()`` (see
+   :ref:`zeek_functions`) from a hook inside your grammar at a point
+   when the parser can be reasonably certain that it is processing the
+   expected protocol. Optionally, you may also call
+   ``zeek::reject_protocol()`` when you're sure the parser is *not*
```

Comment:
Let's call out where one would call these, e.g., `confirm_protocol` in a `%done` hook, and `reject_protocol` in an `%error` hook.

Reply:
I'll add a note on `reject_protocol`. For confirming, there's not much of a common case, depends on there protocol and could be in the middle of some unit.

Reply:
Good point, will do.